ui/src/pages/backoffice/ILL/BorrowingRequest/BorrowingRequestDetails/components/BorrowingRequestPatronLoan/BorrowingRequestPatronLoan.js

```diff
@@ -0,0 +1,267 @@
+          <Grid.Column width={10}>
+            Create a new loan for this borrowed item
+          </Grid.Column>
+          <Grid.Column width={6}>
+            <Button
+              icon
+              positive
+              size="small"
+              labelPosition="left"
+              onClick={this.handleOpenModal}
+            >
+              <Icon name="plus" />
+              New loan
+            </Button>
+          </Grid.Column>
+        </Grid.Row>
+      </Grid>
+    );
+
+    const createLoanModal = (
+      <Modal open={this.state.modalOpen}>
+        <Modal.Header>Create a new loan</Modal.Header>
+        <Modal.Content>
+          Checkout the borrowed item for the patron {brwReq.patron.name}. Select
+          the start and the end dates:
+          <Divider hidden />
+          <Form>
+            <Form.Group>
+              <Form.Field inline required>
+                <label>Start date</label>
+                <DatePicker
+                  minDate={this.today}
+                  defaultValue={this.today}
+                  placeholder="Start date"
+                  handleDateChange={value => this.handleStartDateChange(value)}
+                />
+              </Form.Field>
+              <Form.Field inline required>
+                <label>End date</label>
+                <DatePicker
+                  minDate={this.today}
+                  defaultValue={this.endDate}
+                  placeholder="End date"
+                  handleDateChange={value => this.handleEndDateChange(value)}
+                />
+              </Form.Field>
+            </Form.Group>
+            <Divider hidden />
+            <i>
+              The loan end date should not be after the borrowing request end
+              date.
+            </i>
+          </Form>
+          {hasError && (
+            <Message
+              error
+              header="Something went wrong"
+              content={this.transformError(error)}
+            />
+          )}
+        </Modal.Content>
+        <Modal.Actions key={'modalActions'}>
+          <Button onClick={this.handleCloseModal}>Cancel</Button>
+          <Button
+            positive
+            loading={isLoading}
+            disabled={!this.isSelectionValid() || isLoading}
+            icon="checkmark"
+            labelPosition="left"
+            content="Create"
+            onClick={this.handleCreateLoan}
+          />
+        </Modal.Actions>
+      </Modal>
+    );
+
+    return (
+      <>
+        <Message>
+          <Message.Content>{createLoanButton}</Message.Content>
+        </Message>
+        {createLoanModal}
+      </>
+    );
+  }
+}
+
+CreateLoan.propTypes = {
+  brwReq: PropTypes.object.isRequired,
+  /* REDUX */
+  error: PropTypes.object,
+  isLoading: PropTypes.bool.isRequired,
+  hasError: PropTypes.bool.isRequired,
+  borrowingRequestCreateLoan: PropTypes.func.isRequired,
+};
+
+export default class BorrowingRequestPatronLoan extends React.Component {
```

Comment:
maybe it would make sense at some point to split those components on two separate files

Reply:
Agree, if you don't mind I/we will do it when working on extensions, since it might change the behaviour here. I would go for this splitting now.